board.es6

```diff
@@ -266,30 +224,74 @@ class Board {
     return Board.parseTeam( this.pieceObject(position) )=== Board.WHITE && Board.parseSpecies( this.pieceObject(position) ) === Board.PAWN
   }
 
-  blackPawnDoubleSteppedFrom(position){
-    return this.previousLayouts.length && this.positionEmpty(position) && Board.parseTeam( this.pieceObjectFromLastLayout(position) ) === Board.BLACK && Board.parseSpecies( this.pieceObjectFromLastLayout(position) ) === Board.PAWN
+  blackPawnDoubleSteppedTo(position){// only to be called if already know the black pawn is at rank 4 and in "position"
+    var result;
+    let blackMoves = this.movesNotationFor(Board.BLACK),
+      square = Board.gridCalculator(position),
+      hypotheticalSingleStepSquare = square[0] + '6'; //e.g. if the double step would've been to a4, then the single step it may have taken would've been to a3
+    if( blackMoves[blackMoves.length -1] === square ){
+      result = true;
+    } else {
+      return false //this means that even if it did at some point double step, it wasn't the last move to occur
+    }
+    for(let i = 0; i < blackMoves; i < blackMoves.length){
+      if( blackMoves[i] === hypotheticalSingleStepSquare || blackMoves[i] === hypotheticalSingleStepSquare + "+" ){
+        result = false;
+        break
+      }
+    }
+    return result
   }
 
-  whitePawnDoubleSteppedFrom(position){
-    return this.previousLayouts.length && this.positionEmpty(position) && Board.parseTeam( this.pieceObjectFromLastLayout(position) ) === Board.WHITE && Board.parseSpecies( this.pieceObjectFromLastLayout(position) ) === Board.PAWN
+
+  whitePawnDoubleSteppedTo(position){// only to be called if already know the white pawn is at rank 4 and in "position"
```

Comment:
This and the `blackPawnDoubleSteppedTo` method have very similar logic. I'd combine them into a single method `doubleSteppedFrom` that takes in a moves a and a `hypotheticalSingleStepSquare`.